src/jdns/jdns.c

```diff
@@ -2067,6 +2067,10 @@ int _unicast_do_writes(jdns_session_t *s, int now)
 		}
 
 		// in theory, it is not possible for 'ns' to be null here
+		if (ns == 0)
+		{
+			continue; // but lets make sure it will not stab us in the back
```

Comment:
I had problems with multiple queries on different network infarces (not sure if this is relevant) before I applied the coverity fixes. However, I'm not 100% sure if this was related. Anyway, I think the additional check does not hurt.